service/internal/access/v2/pdp.go

```diff
@@ -227,6 +245,111 @@ func (p *PolicyDecisionPoint) GetDecision(
 	return decision, nil
 }
 
+func (p *PolicyDecisionPoint) GetDecisionRegisteredResource(
+	ctx context.Context,
+	registeredResourceValueFQN string,
+	action *policy.Action,
+	resources []*authz.Resource,
+) (*Decision, error) {
+	l := p.logger.With("registeredResourceValueFQN", registeredResourceValueFQN)
+	l = l.With("action", action.GetName())
+	l.DebugContext(ctx, "getting decision", slog.Int("resourcesCount", len(resources)))
+
+	if err := validateGetDecisionRegisteredResource(registeredResourceValueFQN, action, resources); err != nil {
+		return nil, err
+	}
+
+	registeredResourceValue := p.allRegisteredResourceValuesByFQN[registeredResourceValueFQN]
+	if err := validateRegisteredResourceValue(registeredResourceValue); err != nil {
+		return nil, err
+	}
+
+	// Filter all attributes down to only those that relevant to the entitlement decisioning of these specific resources
+	decisionableAttributes := make(map[string]*attrs.GetAttributeValuesByFqnsResponse_AttributeAndValue)
+
+	for idx, resource := range resources {
+		// Assign indexed ephemeral ID for resource if not already set
+		if resource.GetEphemeralId() == "" {
+			resource.EphemeralId = "resource-" + strconv.Itoa(idx)
+		}
+
+		for idx, valueFQN := range resource.GetAttributeValues().GetFqns() {
+			// lowercase each resource attribute value FQN for case consistent map key lookups
+			valueFQN = strings.ToLower(valueFQN)
+			resource.GetAttributeValues().Fqns[idx] = valueFQN
+
+			// If same value FQN more than once, skip
+			if _, ok := decisionableAttributes[valueFQN]; ok {
+				continue
+			}
+
+			attributeAndValue, ok := p.allEntitleableAttributesByValueFQN[valueFQN]
+			if !ok {
+				return nil, fmt.Errorf("resource value FQN not found in memory [%s]: %w", valueFQN, ErrInvalidResource)
+			}
+
+			decisionableAttributes[valueFQN] = attributeAndValue
+			err := populateHigherValuesIfHierarchy(ctx, p.logger, valueFQN, attributeAndValue.GetAttribute(), p.allEntitleableAttributesByValueFQN, decisionableAttributes)
+			if err != nil {
+				return nil, fmt.Errorf("error populating higher hierarchy attribute values: %w", err)
+			}
+		}
+	}
+	l.DebugContext(ctx, "filtered to only entitlements relevant to decisioning", slog.Int("decisionableAttributeValuesCount", len(decisionableAttributes)))
+
+	entitledFQNsToActions := make(map[string][]*policy.Action)
```

Comment:
Should we abstract this function so we can separately test it as the logic is also used in GetEntitlements?

Reply:
It looks like the current logic including `populateHigherValuesIfHierarchy` aligns with the current attribute value decisioning implementation. I will need to take a deeper look (probably in a sync with you) to answer firmly, but I expect us to have enough unit test scenarios to know clearly if something is missing here by the time test coverage is complete.

Reply:
IIRC, we populate the higher values in a hierarchy attribute to be able to resolve subject mappings, which isn't an issue here when we are not applying subject mappings at all to stored action-attribute-values and instead doing a more direct comparison on the entitlements map we build from them.

Reply:
resolving as stale after talking to @jakedoublev